src/core/namespace.c

```diff
@@ -498,9 +508,19 @@ static int append_extensions(
 
         CLEANUP_ARRAY(overlays, n_overlays, strv_free_many);
 
+        /* Prepare a list of verity metadata, that will be used to check the root hash of each image and
+         * drop duplicates. Overlayfs rejects duplicate lowerdir= even if they are mounted on different
+         * paths, as it resolves each mount to its source device, rather than using the lowerdir. */
```

Comment:
i really don't like the comment. "duplicate lowerdir=" sounds just misleading, because key here is that you might be referencing different images, i.e. different files on disk, but *still* they are considered the same here.

hence make this explicit: overlayfs checks the fs uuid and the file handle (as in name_to_handle_at) of the top-level inode, and that's going to be the same of the root o of all file system images with the same root hashes.

"resolves each mount to its source device"? that's misleading too, overlayfs has no concept of devices. Only of superblocks, and inodes on them.

Reply:
uhm were you looking at the old commit? The current wording is:

"Overlayfs rejects supplying the same filesystem superblock multiple times in lowerdir=, even if they are mounted on different paths, as it resolves each mount to its source filesystem"

Does this work?

Reply:
current wording:

"Overlayfs rejects supplying the same directory inode twice as determined by filesystem UUID and file handle in 
lowerdir=, even if they are mounted on different paths, as it resolves each mount to its source filesystem"

If this doesn't work, please let me know exactly how you want it phrased, and I'll apply it